cpp/src/parquet/arrow/reader.h

```diff
@@ -115,17 +115,32 @@ class RowGroupReader;
 // arrays
 class PARQUET_EXPORT FileReader {
  public:
-  /// Factory function to create a FileReader from a ParquetFileReader and properties
+  // Factory function to create a FileReader from a ParquetFileReader and properties.
+  // \deprecated Deprecated in 19.0.0. Use arrow::Result version instead.
```

Comment:
We should use `///` not `//` for Doxygen: https://www.doxygen.nl/manual/docblocks.html

```suggestion
  /// Factory function to create a FileReader from a ParquetFileReader and properties.
  /// \deprecated Deprecated in 19.0.0. Use arrow::Result version instead.
```